### ISSUE TITLE
Keeping connection to amqp alive to listen for incoming messages

### ISSUE THREAD

**Issue by SeedyROM** [cloudamqp/amqp-client.cr#5]
I think I'm missing something super important *(and none of the docs or source code from this project have enlightened me. I've read every line of the source over the past 2 days)*.

It seems every other client I've found for working with amqp has the ability to invoke a listening state that doesn't just close the connection when the blocks are run.

For instance from the [Hello World (Ruby)](https://www.rabbitmq.com/tutorials/tutorial-one-ruby.html) example from the RabbitMQ documentation clearly shows how to do this:
```ruby
begin
  puts ' [*] Waiting for messages. To exit press CTRL+C'
  queue.subscribe(block: true) do |_delivery_info, _properties, body|
    puts " [x] Received #{body}"
  end
rescue Interrupt => _
  connection.close

  exit(0)
end
```
It's also shown in literally every other example in the tutorial, but I figured Ruby would be a closer base to bring up since talking about Crystal.

However subscribing to the queue with this package instantly closes before any messages can be received from anywhere outside of the local block.

**Am I crazy for thinking it's weird that I can't seem to pull this off?**
**Is there something super fundamental I'm missing?**

**Comment by carlhoerberg** [comment]
ok, no that would be easy to implement. 

you can also just do a `sleep` after the subscribe block

**Comment by carlhoerberg** [comment]
Fixed in v0.4.5

**Comment by SeedyROM** [comment]
@carlhoerberg Just wanted to say thank you for so politely not only helping me figure this out but releasing that new update! Cheers mate!!!
